api/jsonapi.go

```diff
@@ -2501,7 +2501,21 @@ func (i *jsonAPIHandler) GETPurchases(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	offsetId := r.URL.Query().Get("offsetId")
-	purchases, err := i.node.Datastore.Purchases().GetAll(offsetId, l)
+	stateQuery := r.URL.Query().Get("state")
```

Comment:
I would put then in a function like `parseSearchTerms(q url.Values) (string, []pb.OrderState)` and reuse it in the other endpoint.